src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -113,7 +113,10 @@ internal unsafe static bool IsFreeThreaded(IObjectReference objRef)
 
         public static IObjectReference GetObjectReferenceForInterface(IntPtr externalComObject)
         {
-            return GetObjectReferenceForInterface<IUnknownVftbl>(externalComObject, IID.IID_IUnknown);
+            // Here the ptr itself might not point to IUnknown, but we are using IUnknown for the purposes of getting
+            // an agile reference if needed.  Due to that and to keep back compat, making sure to not trigger a QI
+            // In addition, the ptr is already pointing to the correct interface which the IObjectReference is expected for.
+            return GetObjectReferenceForInterface<IUnknownVftbl>(externalComObject, IID.IID_IUnknown, false);

```

Comment:
Question: in theory, like in this case, could we not _always_ skip the QI for `IUnknown`? I mean, since any COM interface will always extend `IUnknown` regardless. As in, unless we specifically want to QI just because we want to test for object identity (in which case `IUnknown` is the only way to do that), isn't that always not needed? Just trying to understand this better.

Reply:
Oh, nono, it was just me double checking that I was getting things right. As in, for the future, can we skip the QI in all cases where don't specifically need to check for object identity, or are there some more gotchas to keep in mind? Basically I'm trying to understand when exactly I'd also have to be careful if I were adding some new code that needed this API, to set that `requireQI` to `true` 🙂